validator/src/utils.rs

```diff
@@ -0,0 +1,95 @@
+use std::fs;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use anchor_client::solana_sdk::ed25519_instruction::{
+    DATA_START, PUBKEY_SERIALIZED_SIZE, SIGNATURE_SERIALIZED_SIZE,
+};
+use anchor_client::solana_sdk::{self};
+use anchor_lang::solana_program::instruction::Instruction;
+use base64::Engine;
+use bytemuck::bytes_of;
+use directories::ProjectDirs;
+
+fn project_dirs() -> ProjectDirs {
+    ProjectDirs::from("com", "GuestChain", "Validator")
+        .unwrap_or_else(|| panic!("Invalid home directory!"))
+}
+
+pub fn config_file() -> PathBuf {
+    let proj_dirs = project_dirs();
+    let mut data_file = PathBuf::from(proj_dirs.config_dir());
+    fs::create_dir_all(&data_file).unwrap();
+    data_file.push("config.toml");
+    data_file
+}
+
+pub fn setup_logging(log_level: &str) {
```

Comment:
This should take log::LevelFilter as argument and parsing should be done by caller.